x/auth/vesting/types/vesting_account_test.go

```diff
@@ -924,3 +925,585 @@ func initBaseAccount() (*authtypes.BaseAccount, sdk.Coins) {
+			rewardCoins:      sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)),
+			expectedVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)), // No change
+			expectedEndTime:  now.Add(24 * time.Hour).Unix(),
+			testTime:         now.Add(12 * time.Hour).Unix(),
+		},
+		{
+			name:             "uneven delegation split (75% vesting, 25% free)",
+			startTime:        now.Unix(),
+			endTime:          now.Add(24 * time.Hour).Unix(),
+			originalVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)),
+			delegatedVesting: sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 75)),
+			delegatedFree:    sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 25)),
+			rewardCoins:      sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)),
+			expectedVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 175)), // Original 100 + 75 new
+			expectedEndTime:  now.Add(24 * time.Hour).Unix(),
+			testTime:         now.Add(12 * time.Hour).Unix(),
+		},
+		{
+			name:             "large reward amount",
+			startTime:        now.Unix(),
+			endTime:          now.Add(24 * time.Hour).Unix(),
+			originalVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)),
+			delegatedVesting: sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 50)),
+			delegatedFree:    sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 50)),
+			rewardCoins:      sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 10000)),
+			expectedVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 5100)), // Original 100 + 5000 new
+			expectedEndTime:  now.Add(24 * time.Hour).Unix(),
+			testTime:         now.Add(12 * time.Hour).Unix(),
+		},
+		{
+			name:             "partial delegation (50% of vesting delegated)",
+			startTime:        now.Unix(),
+			endTime:          now.Add(24 * time.Hour).Unix(),
+			originalVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)),
+			delegatedVesting: sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 50)),
+			delegatedFree:    sdk.NewCoins(),
+			rewardCoins:      sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)),
+			expectedVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 200)), // Original 100 + 100 new
+			expectedEndTime:  now.Add(24 * time.Hour).Unix(),
+			testTime:         now.Add(12 * time.Hour).Unix(),
+		},
+		{
+			name:             "update after vesting period completed",
+			startTime:        now.Add(-48 * time.Hour).Unix(), // Start time in the past
+			endTime:          now.Add(-24 * time.Hour).Unix(), // End time in the past
+			originalVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)),
+			delegatedVesting: sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 50)),
+			delegatedFree:    sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 50)),
+			rewardCoins:      sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 100)),
+			expectedVesting:  sdk.NewCoins(sdk.NewInt64Coin(stakeDenom, 150)), // Original 100 + 50 new
+			expectedEndTime:  now.Add(-24 * time.Hour).Unix(),                 // End time should remain unchanged
+			testTime:         now.Unix(),
```

Comment:
If the end time is in the past, why is delegatedVesting non-zero? and why is the expecting vesting not 100?